platform/bukkit/src/main/java/me/lokka30/treasury/plugin/bukkit/hooks/miniplaceholders/MiniPlaceholdersHook.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * This file is/was part of Treasury. To read more information about Treasury such as its licensing, see <https://github.com/ArcanePlugins/Treasury>.
+ */
+
+package me.lokka30.treasury.plugin.bukkit.hooks.miniplaceholders;
+
+import io.github.miniplaceholders.api.Expansion;
+import java.io.File;
+import java.util.Locale;
+import java.util.UUID;
+import me.lokka30.treasury.plugin.bukkit.BukkitTreasuryPlugin;
+import me.lokka30.treasury.plugin.bukkit.vendor.BukkitVendor;
+import me.lokka30.treasury.plugin.core.hooks.Hook;
+import me.lokka30.treasury.plugin.core.hooks.PlayerData;
+import me.lokka30.treasury.plugin.core.hooks.placeholder.BasicPlaceholderExpansion;
+import me.lokka30.treasury.plugin.core.hooks.placeholder.PlaceholdersConfig;
+import me.lokka30.treasury.plugin.core.hooks.placeholder.minipspecific.MiniPlaceholdersConfig;
+import net.kyori.adventure.identity.Identity;
+import net.kyori.adventure.text.minimessage.tag.Tag;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+public class MiniPlaceholdersHook implements Hook {
+
+    private BasicPlaceholderExpansion basicExpansion;
+
+    @Override
+    public @NotNull String getPlugin() {
+        return "MiniPlaceholders";
+    }
+
+    @Override
+    public boolean register() {
+        if (!BukkitVendor.isPaper() && !BukkitVendor.isFolia()) {
+            return false;
+        }
+        PlaceholdersConfig config = MiniPlaceholdersConfig.load(new File(
+                ((BukkitTreasuryPlugin) BukkitTreasuryPlugin.getInstance()).dataFolder(),
+                "miniPlaceholders_config.yml"
+        ));
+        this.basicExpansion = new BasicPlaceholderExpansion(config);
+        Expansion expansion = Expansion.builder("treasury").globalPlaceholder(
```

Comment:
I would recommend adding a filter for the Player class, so you can make sure that the only Audience type from which you can get placeholders is the Player class.
Also, this ensures that when passing wrapped Audiences based on a player, always use the internal Player object to get the placeholders. This was used in some plugins like [Carbon](https://modrinth.com/plugin/carbon)

This filter is only used for Audience and Relational placeholders, global placeholders are not affected

```suggestion
        Expansion expansion = Expansion.builder("treasury").filter(Player.class).globalPlaceholder(
```